src/api/plugins/tdbp/tests/factory.py

```diff
@@ -0,0 +1,178 @@
+        ).isoformat()
+        actor_name = f"student_{student_id}"
+        action_id = f"https://fake-lms.com/action/{action_id}"
+
+        if event_name in Activities:
+            grade = {"score": {"scaled": uniform(0.0, 20.0)}}
+            return {
+                "timestamp": timestamp,
+                "actor": {
+                    "account": {"name": actor_name, "homePage": "https://fake-lms.com"}
+                },
+                "object": {
+                    "id": action_id,
+                    "definition": {"name": f"Action {action_id}"},
+                },
+                "verb": {"id": "https://xapi.com/fake-verb"},
+                "context": {
+                    "extensions": {
+                        "http://lrs.learninglocker.net/define/extensions/info": {
+                            "event_name": event_name
+                        }
+                    },
+                },
+                "result": grade,
+            }
+
+        else:
+            return {
+                "timestamp": timestamp,
+                "actor": {
+                    "account": {"name": actor_name, "homePage": "https://fake-lms.com"}
+                },
+                "object": {
+                    "id": action_id,
+                    "definition": {"name": f"Action {action_id}"},
+                },
+                "verb": {"id": "https://xapi.com/fake-verb"},
+                "context": {
+                    "extensions": {
+                        "http://lrs.learninglocker.net/define/extensions/info": {
+                            "event_name": event_name
+                        }
+                    }
+                },
+            }
+
+    def _associate_event_names_to_actions(self, event_names: List, action_ids: List):
+        # Shuffle the event names and action indices
+        shuffled_event_names = sample(event_names, len(event_names))
+        shuffled_indices = sample(action_ids, len(action_ids))
+
+        # Create a dictionary with randomly associated values
+        result = dict(zip(shuffled_indices, shuffled_event_names))
+
+        return result
+
+    def generate_statements(self):
+        """Generate a set of statements given window, cohort and actions constraints."""
+        statements = []
+
+        active_actions_event_names = self._associate_event_names_to_actions(
+            [activity.value for activity in Activities],
+            list(range(1, self.settings.ACTIVE_ACTIVITIES + 1)),
+        )
+        active_actions_event_names.update(
+            self._associate_event_names_to_actions(
+                [ressource.value for ressource in Ressources],
+                list(
+                    range(
+                        self.settings.ACTIVE_ACTIVITIES + 1,
+                        self.settings.ACTIVE_ACTIONS + 1,
+                    )
+                ),
+            )
+        )
+
+        for action_id in range(1, self.settings.ACTIVE_ACTIONS + 1):
```

Comment:
I don't understand why you start `action_id` at 1 here. We can't start at 0? It would make code easier to read